SPECS/linstor-client.spec

```diff
@@ -0,0 +1,43 @@
+Summary: DRBD distributed resource management utility
+Name:    linstor-client
+Version: 1.19.0
+Release: 1%{?dist}
+License: GPLv3
+URL:     https://linbit.com/linstor/
+
+BuildArch: noarch
+
+Source0: https://pkg.linbit.com//downloads/linstor/%{name}-%{version}.tar.gz
+
+%define python3_installed %(which python3 2>/dev/null)
+
+# The build container for XCP-ng 8.2 doesn't have python3, so python3 available
+# means we're building for XCP-ng 8.3 and use python3, python2 otherwise
+%if "%(which python3 2>/dev/null)"
+%define python "python3"
+BuildRequires: python3-setuptools
+%else
+%define python "python2"
+BuildRequires: python-setuptools
+%endif
+
+Requires: python-linstor >= 1.19.0
+
+%description
+This client program communicates to controller node which manages the resources
+
+%prep
+%autosetup -p1
+
+%build
+PYTHON=%{python} %{python} ./setup.py build
+
+%install
+PYTHON=%{python} %{python} ./setup.py install --single-version-externally-managed -O1 --root=$RPM_BUILD_ROOT --record=INSTALLED_FILES
+
+%files -f INSTALLED_FILES
```

Comment:
This works, but I see several drawbacks in this:
* you can't tell what the RPM's contents will be by just reading the spec file. You need to have a built RPM to check it.
* as a reviewer,  I don't know if there are files that should be identified with `%doc`, `%config` or other macros with a special meaning.
* I was taught, in my early packaging days, that it's better to be explicit in the list of files or at least directories in the `%files` section, so that any missing file, or unexpected new file, is detected at build time and makes the build fail. This way, every file included in the RPM is a conscious choice from the packager.